Objective-C/CBLDatabase.mm

```diff
@@ -624,29 +632,47 @@ - (BOOL) createIndexWithConfig: (CBLIndexConfiguration*)config
 }
 
 - (BOOL) createIndex: (NSString*)name withConfig: (id<CBLIndexSpec>)config error: (NSError**)error {
-    return [[self mustDefaultCollection: error] createIndexWithName: name
-                                                             config: config
-                                                              error: error];
+    NSError *e = nil;
+    BOOL res = [[self defaultCollectionOrThrow] createIndexWithName: name config: config error: &e];
+    if (error)
```

Comment:
Better to do this after `throwIfNotOpenError(e)` so if the exception is thrown, the error will not be set.